app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/MainActivity.kt

```diff
@@ -22,22 +22,47 @@ import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import com.esri.arcgismaps.kotlin.sampleviewer.navigation.NavGraph
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
 import com.esri.arcgismaps.sample.sampleslib.theme.SampleAppTheme
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enableEdgeToEdge()
         setContent {
-            setContent {
-
-                SampleAppTheme {
-                    Surface(color = MaterialTheme.colorScheme.background) {
-                        NavGraph()
-                    }
+            SampleAppTheme {
+                Surface(color = MaterialTheme.colorScheme.background) {
+                    NavGraph()
+                    HandleExceptions()
                 }
             }
         }
     }
+
+    /**
+     * Responsible for managing incoming exceptions and displaying it as a dialog in the UI.
+     */
+    @Composable
+    fun HandleExceptions() {
+        var sampleViewerException by remember { mutableStateOf(arrayOf<String>()) }
+        LaunchedEffect(Unit) {
+            intent.extras?.let {
+                sampleViewerException = it.getStringArray("SampleViewerException") ?: arrayOf()
+            }
+        }
```

Comment:
This seems like it could be simplified. the `remember` call will run once per composition as will the LaunchedEffect so the logic in the LaunchedEffect could be run in the remember call.

```suggestion
        var sampleViewerException by remember { 
            mutableStateOf(
                intent.extras?.let {
                    it.getStringArray("SampleViewerException") ?: arrayOf()
                } ?: arrayOf<String>()
            ) 
        }
```